### DISCUSSION TITLE
关于播放列表/字幕列表的问题

### DISCUSSION THREAD

**Discussion by 3384117899** [hooke007/mpv_PlayKit#407] [mpv 通用]
大佬，用的代码是来自https://github.com/FinnRaze/mpv-osc-modern-f/tree/main  
就是里面我改了改界面后，想在点击比如字幕或者列表的时候可以出现列表我该怎么可以和你的播放列表字幕列表可以结合一下
下面这上面那个代码里面的
   --cy_sub
    ne = new_element('cy_sub', 'button')
    ne.enabled = (#tracks_osc.sub > 0)
    ne.visible = (osc_param.playresx >= 540)
    ne.content = '\xEF\x8C\xA4'
    ne.tooltip_style = osc_styles.Tooltip
    ne.tooltipF = function ()
        local msg = texts.off
        if not (get_track('sub') == 0) then
            msg = (texts.subtitle .. ' [' .. get_track('sub') .. ' ∕ ' .. #tracks_osc.sub .. '] ')
            local prop = mp.get_property('current-tracks/sub/lang')
            if not prop then
                prop = texts.na
            end
            msg = msg .. '[' .. prop .. ']'
            prop = mp.get_property('current-tracks/sub/title')
            if prop then
                msg = msg .. ' ' .. prop
            end
            return msg
        end
        return msg
    end
    ne.eventresponder['mbtn_left_up'] =
        function () set_track('sub', 1) end
    ne.eventresponder['mbtn_right_up'] =
        function () set_track('sub', -1) end
    ne.eventresponder['shift+mbtn_left_down'] =
        function () show_message(get_tracklist('sub')) end
    ne.eventresponder["wheel_down_press"] =
        function () set_track("sub", 1) end
    ne.eventresponder["wheel_up_press"] =
        function () set_track("sub", -1) end

下面是我改了一点

    --cy_sub-----------------------------------------------------
    ne = new_element('cy_sub', 'button')
    ne.enabled = (#tracks_osc.sub > 0)
    ne.visible = (osc_param.playresx >= 540)
    ne.content = '\xEF\x8C\xA4'
    ne.tooltip_style = osc_styles.Tooltip
    ne.tooltipF = function ()
        local msg = texts.off
        if not (get_track('sub') == 0) then
            msg = (texts.subtitle .. ' [' .. get_track('sub') .. ' ∕ ' .. #tracks_osc.sub .. '] ')
            local prop = mp.get_property('current-tracks/sub/lang')
            if not prop then
                prop = texts.na
            end
            msg = msg .. '[' .. prop .. ']'
            prop = mp.get_property('current-tracks/sub/title')
            if prop then
                msg = msg .. ' ' .. prop
            end
            return msg
        end
        return msg
    end

    ne.eventresponder['mbtn_left_up'] =
        function ()
        local tracklist_table = get_tracklist_table('sub')

        -- 在这里处理表格数据，例如构建一个字符串或者其他你想要的显示方式
        local msg = "Subtitles Tracklist:\n"

        for _, track in ipairs(tracklist_table) do
            local selected = track.selected and '●' or '○'
            msg = msg .. selected .. ' ' .. track.index .. ': [' .. track.lang .. '] ' .. track.title .. '\n'
        end

        show_message(msg)
    end
     
    ne.eventresponder['mbtn_right_up'] =
        function () set_track('sub', -1) end
    ne.eventresponder['shift+mbtn_left_down'] =
        function () show_message(get_tracklist('sub')) end
    ne.eventresponder["wheel_down_press"] =
        function () set_track("sub", 1) end
    ne.eventresponder["wheel_up_press"] =
        function () set_track("sub", -1) end

我设置完点击左键，能显示字幕内容，但就显示两秒就没了，只能看一眼
有没有改的方法，或者和你的playlist  sublist结合一下

**Comment by hooke007** [comment]
https://github.com/hooke007/MPV_lazy/blob/e6a4d79181c51e442d4ce4e97c881c70f2b13479/portable_config/input_uosc.conf#L8-L13

**Reply by hooke007** [reply]
那就换个插件

**Reply by 3384117899** [reply]
哦哦哦，好的
